kvserver/rangefeed: add more tests for StreamManager

wenyihu6 opened the pull request:
Previously, we updated the UnbufferedSender to use
the StreamManager. This patch adds more tests for
StreamManager.

Part of: https://github.com/cockroachdb/cockroach/issues/110432
Release note: none

Co-authored-by: Steven Danna danna@cockroachlabs.com

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135075)
<!-- Reviewable:end -->

wenyihu6 committed:
kvserver/rangefeed: add more tests for StreamManager

Previously, we updated the UnbufferedSender to use
the StreamManager. This patch adds more tests for
StreamManager.

Part of: #110432
Release note: none

Co-authored-by: Steven Danna danna@cockroachlabs.com

tbg reviewed (approved):
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135075)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @stevendanna)



<!-- Sent from Reviewable.io -->

wenyihu6 commented:
TFTR!

bors r=tbg

wenyihu6 commented:
The last push fixed a race condition in the tests that I found during local stress testing.

bors r=tbg
